### PR TITLE
Design system button

### PR DESCRIPTION
**Pull request by petemill**
Introduces:
- Design System button based on the one in Figma. Includes "primary" and "secondary" levels, as well as all sizes from "tiny" to 'jumbo". Does _not_ include "colored" theme version or "tertiary" variants as well as the separate icon-only buttons.
- CSS modules

Fixes button in NTP customize panel, specifically for Brave News RSS feed adding.

Resolves https://github.com/brave/brave-browser/issues/20639

image
image

## Test Plan:
In storybook or in real browser with Brave News enabled:
- Open NTP
- Open Customize panel
- Switch to Brave News tab
- Enter an rss feed url
- Observe button states for light and dark mode:
    - hover
    - focus
    - click
    - loading

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one https://github.com/brave/security/issues/762
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: package.json
Comment (diracdeltas) [inline package.json:337]: just confirming that these 2 should be in devDeps instead of dependencies

**Comment by petemill** [comment]
### Variations

#### Secondary


image

image

image

image


#### Primary
image
image
image
image

File: package.json
Comment (petemill) [reply package.json:337]: well, sass-loader is used in the build system, so probably? Seems to be what some of these other ones are here for.

typescript-plugin-css-modules (sorry for forgetting that one from the security review) isn't even used in the build process, it's only picked up by language servers in IDEs like VSCode to provide intellisense for css classes.

Does that help @diracdeltas ?

**Review by bsclifton — APPROVED** [review]
Will defer to @diracdeltas for the new deps; but the changes here LGTM! Tested out via Storybook 👍

File: components/web-components/button/button.module.scss
Comment (nullhook) [inline components/web-components/button/button.module.scss:58]: should size variant names be similar on all components? i.e `sm, lg, x-lg`

**Review by nullhook — APPROVED** [review]

File: package.json
Comment (diracdeltas) [inline package.json:328]: do we need to pin to an outdated version here? i noticed the latest is v12

File: package.json
Comment (diracdeltas) [inline package.json:355]: as mentioned in sec review i'm more in favor of just adding a new utility function instead of increasing our dependencies

File: package.json
Comment (diracdeltas) [reply package.json:337]: makes sense

**Comment by petemill** [comment]
noplatform error was due to audit but I'm almost certain that's affecting master too. It's from https://github.com/advisories/GHSA-3949-f494-cm99 which we have via the chain:
```
└─┬ @storybook/addon-knobs@6.4.0
  └─┬ @storybook/components@6.4.13
    └─┬ react-syntax-highlighter@13.5.3
      ├── prismjs@1.26.0
      └─┬ refractor@3.5.0
        └── prismjs@1.25.0
```

**Comment by stephendonner** [comment]
Verified `PASSED` using 

Brave | 1.38.32 Chromium: 99.0.4844.51 (Official Build) nightly (x86_64)
-- | --
Revision | d537ec02474b5afe23684e7963d538896c63ac77-refs/branch-heads/4844@{#875}
OS | macOS Version 11.6.3 (Build 20G415)



`Light`

default | hover | focus | focus + hover | click (doesn't seem possible to capture via screenshot) | loading
--------|------|-------|-------|-------|--------------
Screen Shot 2022-03-09 at 11 03 22 AM | Screen Shot 2022-03-09 at 11 03 55 AM | Screen Shot 2022-03-09 at 11 04 44 AM | Screen Shot 2022-03-09 at 11 18 48 AM | `N/A` | Screen Shot 2022-03-09 at 11 09 54 AM


`Dark`

default | hover | focus | focus + hover | click (doesn't seem possible to capture via screenshot) | loading
-------|--------|-------|------|-------|----------
Screen Shot 2022-03-09 at 11 11 49 AM | Screen Shot 2022-03-09 at 11 13 37 AM | Screen Shot 2022-03-09 at 11 17 17 AM | Screen Shot 2022-03-09 at 11 16 02 AM | `N/A` | Screen Shot 2022-03-09 at 11 14 52 AM

**Pull request merged**
